docs/adr/00xx-moving-business-logic-out-of-controllers.md

```diff
@@ -0,0 +1,124 @@
+# XX. Moving business logic out of controllers
+
+Date: 2019-06-03
+
+## Context
+
+Over time the controller classes for Content Publisher have become
+[increasingly complex][complex-controllers] as the application grew in size
+and functionality. This has led to action methods that are long, difficult to
+understand and expensive to test. Looking towards the future it can be
+anticipated that there will be more functionality in controllers, such as
+access limiting, concurrent editing protection and invalid state handling.
+Therefore there is an expectation that controllers would become an increasing
+pain point for the application due to their complexity.
+
+This is a common scenario for a growing Rails application. A customary
+approach to address it is to distinguish between business and rendering
+logic in an action - where business logic is the action the user requested,
+such as changing application state, and rendering logic is the process of
+building the response to be returned to the user - and move the business logic
+to another class.
+
+Within the Content Publisher team we have been evaluating
+a number of approaches to perform our business logic operations, such
+as [service objects][],
+[the dry-transaction gem][dry-transaction],
+[the interactor gem][interactor-gem],
+[trailblazer operations][trailblazer-operation] and
+the [Publishing API command pattern][pub-api-command]. We came to the
+following conclusions:
+
+- Service objects: This is a pattern already used in Content Publisher,
+  however it is used to perform distinct supporting tasks rather than being
+  coupled to particular controller actions. We felt the services directory had
+  become a mess in Content Publisher, with little consistency between service
+  objects.
+- Dry-transaction gem: This provides a nice interface for managing response
+  through [ruby blocks][calling-a-dry-transaction]. However dry-transaction
+  requires a strict adherence to a provided DSL which can make a transaction
+  class very different from a plain old Ruby object.
+- Interactor gem: This is a simple, flexible pattern which relies on
```

Comment:
This is a nice summary, but it seems a little low-level in comparison to the others. It might be more helpful to contrast it to the other DSL approaches (i.e. focus on the 'flexible' bit) e.g. 'This is a simple, relatively freeform pattern (no DSL) that uses a `context` object for basic control flow'.

Reply:
Done